data/transactions/keyreg_test.go

```diff
@@ -85,8 +85,16 @@ func TestKeyregApply(t *testing.T) {
 	_, err = tx.Apply(mockBal, SpecialAddresses{FeeSink: feeSink}, 0)
 	require.NoError(t, err)
 
-	// Nonparticipatory accounts should not be able to change status
-	mockBal.addrs[src] = basics.BalanceRecord{Addr: src, AccountData: basics.AccountData{Status: basics.NotParticipating}}
-	_, err = tx.Apply(mockBal, SpecialAddresses{FeeSink: feeSink}, 0)
-	require.Error(t, err)
+	// Going from online to nonparticipatory should be okay, if the protocol supports that
+	if mockBal.ConsensusParams().SupportBecomeNonParticipatingTransactions {
```

Comment:
Probably we should change mockBal's version from ConsensusCurrentVersion to ConsensusFuture for this test -- otherwise we aren't testing anything.